compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPeelForLoop.cpp

```diff
@@ -42,17 +83,39 @@ void AMDAIEPeelForLoopPass::runOnOperation() {
       return;
 
     scf::ForOp result;
-    LogicalResult status =
-        scf::peelForLoopFirstIteration(rewriter, forOp, result);
-    if (failed(status)) {
-      return signalPassFailure();
+    switch (peelLoopType) {
+      case PeelLoopType::First:
+        if (failed(scf::peelForLoopFirstIteration(rewriter, forOp, result))) {
+          forOp->emitOpError("failed to peel the first iteration.");
+          return signalPassFailure();
+        }
+        break;
+      case PeelLoopType::Last:
+        if (failed(peelForLoopLastIteration(rewriter, forOp, result))) {
+          forOp->emitOpError("failed to peel the last iteration.");
+          return signalPassFailure();
+        }
+        break;
+      case PeelLoopType::FirstLast:
+        if (failed(scf::peelForLoopFirstIteration(rewriter, forOp, result))) {
+          forOp->emitOpError("failed to peel the first iteration.");
+          return signalPassFailure();
+        }
+        if (failed(peelForLoopLastIteration(rewriter, forOp, result))) {
+          LLVM_DEBUG(llvm::dbgs() << "Skip peeling for the last iteration.\n");
```

Comment:
Here it doesn't send out signalPassFailure() is because we don't want to fail the pass if there are exactly two iterations of the loop. In this case, we don't actually run `peelForLoopLastIteration`,  since the remaining is the last iteration.